spec/core/types.md

```diff
@@ -0,0 +1,466 @@
+# Types
+
+Here we describe the types used in the API.
+
+Some types are defined as _raw_, usually in the form `Raw[TYPE-NAME]`. These types are used for communication between the client and server over JSON-RPC. Each property of a raw type must be a JSON primitive data types as described by the JSON-RPC 2.0 [specification](https://www.jsonrpc.org/specification#parameter_structures) (i.e: Strings, Numbers, Booleans, or Null).
+
```

Comment:
Should we had a table of content so that we can directly jump to a type ?

Reply:
Yes 👍

Reply:
TOC added in [99f04a4](https://github.com/LedgerHQ/wallet-api/pull/8/commits/99f04a4a4e4fd1f8bb66b2042a63907dda2c52d7)

FYI, I used [this extension](https://marketplace.visualstudio.com/items?itemName=yzhang.markdown-all-in-one#table-of-contents) to automatically and reliably generate it 👍